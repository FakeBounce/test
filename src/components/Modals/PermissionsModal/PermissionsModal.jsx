import React, { Component } from "react";
import { translate } from "common/methods/translations";
import CollaborativeSpaceStore from "stores/CollaborativeSpaceStore";
import Select from "elements/Select/Select";

import Modal from "elements/Modal/Modal";
import styles from "./PermissionsModal.scss";
import { Input, Button, Media } from "reactstrap";
import Delete from "assets/images/myEspaceCo/delete.svg";

const permissionsTranslation = {
  r: translate("myCollaborativeSpace.modals.readOnly"),
  rw: translate("myCollaborativeSpace.modals.writeAndRead"),
  rwx: translate("myCollaborativeSpace.modals.fullControl")
};

const permissionsArray = [
  { name: translate("myCollaborativeSpace.modals.readOnly"), value: "r" },
  { name: translate("myCollaborativeSpace.modals.writeAndRead"), value: "rw" },
  { name: translate("myCollaborativeSpace.modals.fullControl"), value: "rwx" }
];

class PermissionsModal extends Component {
  state = {
    currentFile: {},
    usersPermitted: [],
    inheritance: false,
    isAddingUser: false,
    formEmail: "",
    formLevel: ""
  };

  componentWillReceiveProps(nextProps, nextContext) {
    if (nextProps.file !== null) {
      this.getFileInformations(nextProps.file);
      this.setState(state => ({
        ...state,
        currentFile: nextProps.file
      }));
    }
  }

  breakInheritance = async () => {
    const { currentFile } = this.state;
    await CollaborativeSpaceStore.breakFilePermissions(currentFile.id);
    this.setState(state => ({
      ...state,
      inheritance: false
    }));
  };

  handleInputChange = key => e => {
    const val = e.target.value;
    this.setState(state => ({
      ...state,
      [key]: val
    }));
  };

  handleSelectChange = key => e => {
    this.setState(state => ({
      ...state,
      [key]: e
    }));
  };

  toggleAddingUser = bool => () => {
    this.setState(state => ({
      ...state,
      isAddingUser: bool
    }));
  };

  deleteUserPermission = user => async () => {
    const { currentFile } = this.state;
    await CollaborativeSpaceStore.deleteFilePermissions(
      currentFile.id,
      user.email
    );
    this.getFileInformations(currentFile);
  };

  getFileInformations = async file => {
    const permissions = await CollaborativeSpaceStore.getFilePermissions(
      file.id
    );
    let inheritance = false;
    let usersPermitted = [];
    permissions &&
      permissions.map(permission => {
        if (!inheritance && permission.file.permissionInheritance) {
          inheritance = true;
        }
        usersPermitted.push({
          email: permission.principalId,
          level: permission.level
        });
        return null;
      });
    this.setState(state => ({
      ...state,
      usersPermitted,
      inheritance
    }));
    return null;
  };

  addFilePermissions = async () => {
    const { currentFile, formEmail, formLevel } = this.state;
    const { toggle } = this.props;
    await CollaborativeSpaceStore.addFilePermissions(
      currentFile.id,
      formEmail,
      formLevel
    );
    this.setState(state => ({
      ...state,
      usersPermitted: [],
      inheritance: false,
      isAddingUser: false,
      formEmail: "",
      formLevel: ""
    }));
    toggle();
    return null;
  };

  buttons = () => {
    const { isAddingUser } = this.state;
    if (isAddingUser) {
      return (
        <div>
          <Button color="secondary" onClick={this.toggleAddingUser(false)}>
            {translate("globals.cancel")}
          </Button>
          <Button color="primary" onClick={this.addFilePermissions} className={styles.rightButton}>
            {translate("globals.save")}
          </Button>
        </div>
      );
    }
    return null;
  };

  render() {
    const { usersPermitted, isAddingUser, formEmail, inheritance } = this.state;
    return (
      <Modal
        {...this.props}
        className={"NewFileModal"}
        ModalFooterContent={this.buttons}
      >
        <p className={styles.Title}>
          {translate("myCollaborativeSpace.modals.permissions")}
        </p>

        <div className={styles.permissionsActions}>
          <div
            className={styles.addUserPermission}
            onClick={this.toggleAddingUser(!isAddingUser)}
          >
            <span> {isAddingUser ? "-" : "+"}</span>
          </div>
          {inheritance && (
            <div
              className={styles.breakInheritance}
              onClick={this.breakInheritance}
            >
              <span>
                {translate("myCollaborativeSpace.modals.breakInheritance")}
              </span>
            </div>
          )}
        </div>

        {isAddingUser ? (
          <div>
            <p className={styles.inputLabel}>
              {translate("myCollaborativeSpace.user")}
            </p>
            <Input
              placeholder={translate("myCollaborativeSpace.userEmail")}
              value={formEmail}
              onChange={this.handleInputChange("formEmail")}
            />
            <p className={styles.secondInputLabel}>
              {translate("myCollaborativeSpace.userPermissions")}
            </p>
            <Select
              initialOption={""}
              selectedOption={this.handleSelectChange("formLevel")}
              setOptions={permissionsArray}
              className={styles.levelSelect}
            />
          </div>
        ) : (
          usersPermitted.map(userPermission => {
            return (
              <div className={styles.userRow}>
                {userPermission.email}
                {permissionsTranslation[userPermission.level]}
                <div
                  className={styles.DeleteUserPermissions}
                  onClick={this.deleteUserPermission(userPermission)}
                >
                  <Media src={Delete} />
                </div>
              </div>
            );
          })
        )}
        {inheritance && (
          <div className={styles.hasInheritance}>
            {translate("myCollaborativeSpace.modals.hasInheritance")}
          </div>
        )}
      </Modal>
    );
  }
}

export default PermissionsModal;
