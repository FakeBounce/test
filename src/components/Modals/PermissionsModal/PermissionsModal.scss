.Title {
  font-size: 16px;
  color: #1f2b3d;
  margin: 0px;
}
.userRow {
  font-size: 14px;
  color: #1f2b3d;
  width: 90%;
  margin-left: 5%;
  height: 30px;
  justify-content: space-around;
}
.inputLabel {
  font-size: 14px;
  color: #1f2b3d;
  margin-bottom: 15px;
}
.secondInputLabel {
  font-size: 14px;
  color: #1f2b3d;
  margin-top: 15px;
}
.permissionsActions {
  height: 40px;
  width: 100%;
  align-items: center;
  justify-content: center;
  padding: 5px;
  text-align: center;
}
.addUserPermission {
  height: 30px;
  width: 30px;
  border-radius: 5px;
  background-color: #4a90e2;
  align-items: center;
  justify-content: center;
  display: inline-block;
  cursor: pointer;
  text-align: center;
  span {
    color: white;
    font-size: 18px;
  }
}
.breakInheritance {
  height: 30px;
  width: 110px;
  border-radius: 5px;
  background-color: #4a90e2;
  align-items: center;
  justify-content: center;
  display: inline-block;
  cursor: pointer;
  text-align: center;
  font-size: 18px;
  margin-left: 15px;
  span {
    color: white;
    font-size: 12px;
  }
}
.DeleteUserPermissions {
  height: 30px;
  width: 30px;
  border-radius: 5px;
  background-color: #4a90e2;
  align-items: center;
  justify-content: center;
  display: inline-block;
  padding: 3px;
  cursor: pointer;
  text-align: center;
}
.hasInheritance {
  margin-top: 30px;
  margin-bottom: 0px;
  font-size: 14px;
  color: #1f2b3d;
  font-style: italic;
}
.rightButton {
  margin-left: 15px;
}
.levelSelect {
  font-size: 12px;
}
