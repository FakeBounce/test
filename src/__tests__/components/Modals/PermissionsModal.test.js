import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import PermissionsModal from "components/Modals/PermissionsModal/PermissionsModal";

configure({ adapter: new Adapter() });

const defaultTestingProps = {
  isOpen: false,
  title: "Title",
  toggle: () => jest.fn(),
  file: null
};

describe("Testing PermissionsModal component", () => {
  it("renders as expected without parameters", () => {
    const wrapper = shallow(<PermissionsModal />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as closed)", () => {
    const wrapper = shallow(<PermissionsModal {...defaultTestingProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders as expected with parameters (as open)", () => {
    const wrapper = shallow(
      <PermissionsModal {...defaultTestingProps} isOpen />
    );
    expect(wrapper).toMatchSnapshot();
  });

  // @TODO Add tests for functions
});
