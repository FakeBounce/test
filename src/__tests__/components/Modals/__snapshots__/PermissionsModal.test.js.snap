// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PermissionsModal component renders as expected with parameters (as closed) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PermissionsModal
    file={null}
    isOpen={false}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <p
          className={undefined}
>
          Permissions
</p>,
        <div
          className={undefined}
>
          <div
                    className={undefined}
                    onClick={[Function]}
          >
                    <span>
                               
                              +
                    </span>
          </div>
</div>,
        Array [],
        false,
      ],
      "className": "NewFileModal",
      "file": null,
      "isOpen": false,
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Permissions",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Permissions",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
              onClick={[Function]}
>
              <span>
                             
                            +
              </span>
</div>,
            false,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                 
                +
</span>,
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "+",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "+",
              ],
              "type": "span",
            },
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <p
            className={undefined}
>
            Permissions
</p>,
          <div
            className={undefined}
>
            <div
                        className={undefined}
                        onClick={[Function]}
            >
                        <span>
                                     
                                    +
                        </span>
            </div>
</div>,
          Array [],
          false,
        ],
        "className": "NewFileModal",
        "file": null,
        "isOpen": false,
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Permissions",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Permissions",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className={undefined}
                onClick={[Function]}
>
                <span>
                                 
                                +
                </span>
</div>,
              false,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                   
                  +
</span>,
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "+",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "+",
                ],
                "type": "span",
              },
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing PermissionsModal component renders as expected with parameters (as open) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PermissionsModal
    file={null}
    isOpen={true}
    title="Title"
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <p
          className={undefined}
>
          Permissions
</p>,
        <div
          className={undefined}
>
          <div
                    className={undefined}
                    onClick={[Function]}
          >
                    <span>
                               
                              +
                    </span>
          </div>
</div>,
        Array [],
        false,
      ],
      "className": "NewFileModal",
      "file": null,
      "isOpen": true,
      "title": "Title",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Permissions",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Permissions",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
              onClick={[Function]}
>
              <span>
                             
                            +
              </span>
</div>,
            false,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                 
                +
</span>,
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "+",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "+",
              ],
              "type": "span",
            },
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <p
            className={undefined}
>
            Permissions
</p>,
          <div
            className={undefined}
>
            <div
                        className={undefined}
                        onClick={[Function]}
            >
                        <span>
                                     
                                    +
                        </span>
            </div>
</div>,
          Array [],
          false,
        ],
        "className": "NewFileModal",
        "file": null,
        "isOpen": true,
        "title": "Title",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Permissions",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Permissions",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className={undefined}
                onClick={[Function]}
>
                <span>
                                 
                                +
                </span>
</div>,
              false,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                   
                  +
</span>,
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "+",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "+",
                ],
                "type": "span",
              },
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing PermissionsModal component renders as expected without parameters 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PermissionsModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ModalFooterContent": [Function],
      "children": Array [
        <p
          className={undefined}
>
          Permissions
</p>,
        <div
          className={undefined}
>
          <div
                    className={undefined}
                    onClick={[Function]}
          >
                    <span>
                               
                              +
                    </span>
          </div>
</div>,
        Array [],
        false,
      ],
      "className": "NewFileModal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Permissions",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Permissions",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
              onClick={[Function]}
>
              <span>
                             
                            +
              </span>
</div>,
            false,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                 
                +
</span>,
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "+",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "+",
              ],
              "type": "span",
            },
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ModalFooterContent": [Function],
        "children": Array [
          <p
            className={undefined}
>
            Permissions
</p>,
          <div
            className={undefined}
>
            <div
                        className={undefined}
                        onClick={[Function]}
            >
                        <span>
                                     
                                    +
                        </span>
            </div>
</div>,
          Array [],
          false,
        ],
        "className": "NewFileModal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Permissions",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Permissions",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className={undefined}
                onClick={[Function]}
>
                <span>
                                 
                                +
                </span>
</div>,
              false,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                   
                  +
</span>,
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "+",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "+",
                ],
                "type": "span",
              },
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
